clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy

```diff
@@ -97,28 +91,34 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
       credentials.apiAdaptor.hardDestroyReplicaSet(namespace, description.serverGroupName)
     }
 
+    destroyController(credentials, namespace, description.serverGroupName, autoscalerName)
+
     task.updateStatus BASE_PHASE, "Successfully destroyed server group $description.serverGroupName."
   }
 
-  void destroyController(KubernetesV1Credentials credentials, String namespace, String serverGroupName, String autoscalerName) {
-    switch(description.kind) {
+  void destroyController(KubernetesV1Credentials credentials, String namespace, String controllerName, String autoscalerName) {
+    def controllerKind = description.kind
+    switch (controllerKind) {
       case KubernetesUtil.CONTROLLERS_STATEFULSET_KIND:
         if (credentials.apiClientAdaptor.getAutoscaler(namespace, autoscalerName)) {
           task.updateStatus BASE_PHASE, "Destroying autoscaler..."
           if (!credentials.clientApiAdaptor.deleteAutoscaler(namespace, autoscalerName, null, null, null)) {
             throw new KubernetesOperationException("Failed to delete associated autoscaler $autoscalerName in $namespace.")
           }
         }
-        credentials.apiClientAdaptor.hardDestroyStatefulSet(serverGroupName, namespace, null, null, null)
+        credentials.apiClientAdaptor.hardDestroyStatefulSet(controllerName, namespace, null, null, null)
         break
       case KubernetesUtil.CONTROLLERS_DAEMONSET_KIND:
-        credentials.apiClientAdaptor.hardDestroyDaemonSet(serverGroupName, namespace, null, null, null)
+        credentials.apiClientAdaptor.hardDestroyDaemonSet(controllerName, namespace, null, null, null)
         break
       default:
-        throw new KubernetesOperationException("Controller type $description.kind is not support.")
+        if (credentials.clientApiAdaptor.getControllerKind(controllerName, namespace, KubernetesUtil.CONTROLLERS_STATEFULSET_KIND)) {
```

Comment:
Delete can now delete up to three server groups at once? I don't think this is a good practice

Reply:
this is mainly to deal with smoke test framework which use one call to delete all controller.